app/services/images/optimizer.rb

```diff
@@ -53,7 +53,14 @@ def self.translate_cloudinary_options(options)
     end
 
     def self.imgproxy_enabled?
-      Imgproxy.config.key.present? && Imgproxy.config.salt.present?
+      if Rails.env.production?
```

Comment:
as we're trying to remove `if Rails.env.production?` around the code as a general principle, see https://github.com/forem/forem/pull/11096, is there a way for us to not add this check? simply check `(key && salt) || env_var`, would that suffice?

Reply:
Yeah I was reluctant to have this weird requirement either but your suggestion is a good middle ground.

Reply:
Oh second thought, I want to be able to offer a simple setup in development while also prevent accidental release without key and salt on Production. Unlike other env var that accept literally any string, these need to hex-encoded or Imgproxy won't start. Checking only `(key && salt) || env_var` isn't strict enough. If it alright, I want to allow this area to use `Rails.env.production?`

Reply:
> these need to hex-encoded or Imgproxy won't start.

What happens if the key and salt are not properly encoded? Does the app fail to boot? Does it crash when you try to upload a picture?

Reply:
The rail's side would be fine and boot normally. The imgproxy server has to have hex key or it refuses to boot with
```
FATAL   [2020-10-27T19:13:25Z] IMGPROXY_KEY expected to be hex-encoded strings. Invalid: optional
```
If imgproxy server didn't start or wrong/mismatch keys were used in either place, the web app would not be able to render pictures.

Reply:
> Checking only (key && salt) || env_var isn't strict enough :(

The key and salt are going to be configured variables which means they are not going to change often so once they are set the app should be in a pretty good spot. If for some reason they get changed incorrectly and the sign is broken images, I think that is something we can live with. 

Having a production only code path really limits us in our ability to test and run things locally. I think having a production only code path poses a bigger risk for problems down the road with production than having a loose requirement of key/secret presence.

Reply:
Okay, I am undoing my change and keeping it only 1 code path, requiring key/salt in all environment.